Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -46,6 +46,7 @@ def cmd_output(*args)
       ENV["HOMEBREW_BREW_FILE"] = HOMEBREW_PREFIX/"bin/brew"
       ENV["HOMEBREW_INTEGRATION_TEST"] = cmd_id_from_args(args)
       ENV["HOMEBREW_TEST_TMPDIR"] = TEST_TMPDIR
+      ENV["HOMEBREW_NO_GITHUB_API"] = "1"
```

Comment:
Agreed; this makes more sense. I mainly wanted to avoid special casing the integration tests (even though they are currently the only ones that do require a network connection).

:+1: on setting `HOMEBREW_NO_GITHUB_API` in `cmd/tests.rb` unless `--online` is passed (also better aligns with `brew audit` where we also use `--online`).